File: dev-docs/snippets/events/blobs-upload-progress.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires repeatedly as a queued blob upload makes progress.
export function blobsUploadProgress(client: JsBaoClient) {
  // #region example
  client.on("blobs:upload-progress", (payload) => {
    console.log(payload.blobId, payload.status, `${payload.numBytes} bytes`);
  });
  // #endregion example
}

File: dev-docs/snippets/events/blobs-upload-progress.swift
import JsBaoClient

// Fires repeatedly as a queued blob upload makes progress.
func blobsUploadProgress(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.blobsUploadProgress) { (event: BlobUploadProgressEvent) in
    // Full queue-item record, matching JS (queueId/filename/contentType/numBytes/status/attempts/…).
    print(event.blobId, event.filename, "\(event.numBytes) bytes, status \(event.status)")
  }
  // #endregion example
  _ = sub
}
